packages/cli-repl/src/run.ts

```diff
@@ -23,14 +23,14 @@ import { generateUri } from '@mongosh/service-provider-server';
         await runSmokeTests(smokeTestServer, process.execPath, process.argv[1]);
       }
     } else {
-      let mongocryptdSpawnPath = null;
+      let mongocryptdSpawnPaths = [['mongocryptd']];
       if (process.execPath === process.argv[1]) {
         // Remove the built-in Node.js listener that prints e.g. deprecation
         // warnings in single-binary release mode.
         process.removeAllListeners('warning');
         // Look for mongocryptd in the locations where our packaging would
         // have put it.
-        mongocryptdSpawnPath = await getMongocryptdPath();
+        mongocryptdSpawnPaths = await getMongocryptdPaths();
```

Comment:
Can't we bury this path discovery mechanism inside `MongoCryptdManager.start` and hide it from the rest?

Reply:
So … we could, but I still feel like this makes sense in terms of separation of concerns + testability (it’s much easier to test this when we can pass the spawn paths to the MongocryptdManager/CliRepl) – we could probably make the result of `getMongocryptdPaths()` the default there and keep the option for overriding for testing, but I’m not sure if that’s really cleaner